test/orm/test_joins.py

```diff
@@ -2231,6 +2231,19 @@ def test_on_clause_no_right_side(self):
             sess.query(User).join(User.id == Address.user_id)._compile_context,
         )
 
+    def test_on_clause_no_right_side_two(self):
+        User = self.classes.User
+        Address = self.classes.Address
+        sess = create_session()
+
+        right = Address.user_id
+
+        assert_raises_message(
+            sa_exc.ArgumentError,
+            "Join target %s does not refer to a mapped entity" % right,
+            sess.query(User).join(Address.user_id)._compile_context,
```

Comment:
**mike bayer** (zzzeek) wrote:

FYI in 1.3.x, this error would be raised when join() is called

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2099